src/foam/nanos/ruler/test/RulerDAOTest.java

```diff
@@ -88,7 +88,7 @@ public void testRuleHistory(X x) {
 
     // wait for async
     try {
-      Thread.sleep(asyncWait + 100);
+      Thread.sleep(asyncWait + 2000);
```

Comment:
Thinking perhaps a better option is to loop for max amount of time.   so loop, delay 100, for a max of 2000 for example.

Reply:
Looks like you will not get out of the loop, unless you finish all iterations.
![Screen Shot 2020-02-05 at 2 27 45 PM](https://user-images.githubusercontent.com/58699954/73875750-c555a100-4823-11ea-9234-47f9f57873a2.png)

Reply:
move the find into the loop, exit early if not null.